### ISSUE TITLE
MySQL db migration fails (260.6 -> 261)

### ISSUE THREAD

**Issue by x6j8x** [cloudfoundry/bosh#1588]
Mysql2::Error: Cannot change column 'variable_set_id': used in a foreign key constraint 'instances_ibfk_3'

It seems that MySQL doesn't like column modification when there's a foreign key contraint associated with that column. 

The [migration set](https://github.com/cloudfoundry/bosh/blob/a4e847f737343a6c0e86864a758f2175195c9ebe/src/bosh-director/db/migrations/director/20170203212124_add_variables.rb) first adds the foreign key

```ruby
alter_table(:instances) do
     add_foreign_key :variable_set_id, :variable_sets, :null => true
end
```

and later tries to set the column to "not null"

```ruby
alter_table(:instances) do
     set_column_not_null :variable_set_id
end
```

**Comment by cppforlife** [comment]
Interesting that our mysql tests are not catching this. What is your mysql
version?

**Comment by x6j8x** [comment]
Server version:		5.7.11-log MySQL Community Server (GPL)

**Comment by cppforlife** [comment]
Story: https://www.pivotaltracker.com/story/show/140011419

**Comment by cppforlife** [comment]
@x6j8x would you mind sharing a bit more about how you are installing your server. or any particular my.cnf settings we should set. we cannot seem to reproduce the problem.

**Comment by x6j8x** [comment]
It's an out-of-the box MySQL 5.7 RDS instance with standard parameter set.

**Comment by cppforlife** [comment]
@x6j8x we were able to reproduce the problem. we will shortly release 261.2 bosh release with the fix and improve our testing procedure.

**Comment by cppforlife** [comment]
We believe we fixed it in 261.2

**Comment by x6j8x** [comment]
I had to manually remove the remains of the failed migration (variable_sets & variables table), but now it works like a charm! 👍
